src/nvim/os/env.c

```diff
@@ -243,19 +243,24 @@ void expand_env_esc(char_u *srcp, char_u *dst, int dstlen, bool esc, bool one,
         // Verify that we have found the end of a UNIX ${VAR} style variable
         if (src[1] == '{' && *tail != '}') {
           var = NULL;
-        } else if (src[1] == '{') {
-          ++tail;
-        }
+        } else {
+          if (src[1] == '{') {
+            ++tail;
+          }
 #elif defined(MSWIN)
```

Comment:
Far as I can tell, the windows specific control was never introduced into the neovim codebase.

Even in the first commit, [this function](https://github.com/neovim/neovim/blob/72cf89bce8e4230dbc161dc5606f48ef9884ba70/src/misc1.c#L3201) has no additional windows specific code.